12/umbraco-commerce/reference/storefront-api/README.md

```diff
@@ -0,0 +1,611 @@
+
+Prices in Umbraco Commerce can contain a lot of meta data, such as listing applied discounts/adjustments and all values including or excluding those discounts. These are useful when displaying a full order breakdown, but when only needing to present an orders total value, are not necesarry. By default, the Storefront API will only return the final `value` property of a price object, but should you wish to receive this full pricing breakdown you can pass the `$prices` shortcut key to expand all price values to include this extra meta data.
+
+**Request**
+
+```http
+GET /umbraco/commerce/storefront/api/v1/order/af697207-d370-4aee-824c-15711d43a9f2?expand=$prices
+```
+
+**Response**
+
+```json
+{
+    ...
+    "totalPrice": {
+        "previousAdjustments": {
+            "currency": {
+                "$type": "CurrencyRef",
+                "code": "GBP",
+                "id": "30b62176-6a9e-4a51-b2f0-7ce6c80a461a"
+            },
+            "formatted": {
+                "tax": "-£0.45",
+                "withTax": "-£2.60",
+                "withoutTax": "-£2.15"
+            },
+            "tax": -0.45,
+            "withTax": -2.60,
+            "withoutTax": -2.15
+        },
+        "totalAdjustment": {
+            "currency": {
+                "$type": "CurrencyRef",
+                "code": "GBP",
+                "id": "30b62176-6a9e-4a51-b2f0-7ce6c80a461a"
+            },
+            "formatted": {
+                "tax": "-£0.45",
+                "withTax": "-£2.60",
+                "withoutTax": "-£2.15"
+            },
+            "tax": -0.45,
+            "withTax": -2.60,
+            "withoutTax": -2.15
+        },
+        "value": {
+            "currency": {
+                "$type": "CurrencyRef",
+                "code": "GBP",
+                "id": "30b62176-6a9e-4a51-b2f0-7ce6c80a461a"
+            },
+            "formatted": {
+                "tax": "£5.80",
+                "withTax": "£33.40",
+                "withoutTax": "£27.60"
+            },
+            "tax": 5.80,
+            "withTax": 33.40,
+            "withoutTax": 27.60
+        },
+        "withPreviousAdjustments": {
+            "currency": {
+                "$type": "CurrencyRef",
+                "code": "GBP",
+                "id": "30b62176-6a9e-4a51-b2f0-7ce6c80a461a"
+            },
+            "formatted": {
+                "tax": "£5.80",
+                "withTax": "£33.40",
+                "withoutTax": "£27.60"
+            },
+            "tax": 5.80,
+            "withTax": 33.40,
+            "withoutTax": 27.60
+        }
+    },
+    ...
+}
+```
+
+</details>
+
+<details>
+
+<summary>Fields</summary>
+
+A common pitfall of REST APIs is the problem of over-fetching, which is where an endpoint returns more information that you need. To help with this the Storefront API supports the passing of a `fields` which allows you to define exactly which properties of an object you wish to return, reducing the payload size.
```

Comment:
```suggestion
A common pitfall of REST APIs is the problem of over-fetching, which is where an endpoint returns more information that you need. The Storefront API supports the passing of `fields` which allows you to define exactly which properties of an object you wish to return. This will reduce the payload size.
```